pennylane/beta/interfaces/autograd.py

```diff
@@ -0,0 +1,164 @@
+    tape **in place**.
+
+    Once created, the autograd interface can be used to perform quantum-classical
+    differentiable programming.
+
+    .. note::
+
+        If using a device that supports native autograd computation and backpropagation, such as
+        :class:`~.DefaultQubitAutograd`, the Autograd interface **does not need to be applied**. It
+        is only applied to tapes executed on non-Autograd compatible devices.
+
+    **Example**
+
+    Once an autograd quantum tape has been created, it can be differentiated using autograd:
+
+    .. code-block:: python
+
+        tape = AutogradInterface.apply(QuantumTape())
+
+        with tape:
+            qml.Rot(0, 0, 0, wires=0)
+            expval(qml.PauliX(0))
+
+        def cost_fn(x, y, z, device):
+            tape.set_parameters([x, y ** 2, y * np.sin(z)], trainable_only=False)
+            return tape.execute(device=device)
+
+    >>> x = np.array(0.1, requires_grad=False)
+    >>> y = np.array(0.2, requires_grad=True)
+    >>> z = np.array(0.3, requires_grad=True)
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> cost_fn(x, y, z, device=dev)
+    [0.03991951]
+    >>> jac_fn = qml.jacobian(cost_fn)
+    >>> jac_fn(x, y, z, device=dev)
+    [[ 0.39828408 -0.00045133]]
+    """
+
+    # pylint: disable=attribute-defined-outside-init
+    dtype = np.float64
+
+    @property
+    def interface(self):  # pylint: disable=missing-function-docstring
+        return "autograd"
+
+    def _update_trainable_params(self):
+        params = []
+
+        for p_idx in self._par_info:
+            op = self._par_info[p_idx]["op"]
+            op_idx = self._par_info[p_idx]["p_idx"]
+            params.append(op.data[op_idx])
+
+        trainable_params = set()
+
+        for idx, p in enumerate(params):
+            if getattr(p, "requires_grad", True):
+                trainable_params.add(idx)
+
+        self.trainable_params = trainable_params
+        return params
+
+    def get_parameters(self, trainable_only=True):  # pylint: disable=missing-function-docstring
+        params = self._update_trainable_params()
+
+        if trainable_only:
+            params = [p for idx, p in enumerate(params) if idx in self.trainable_params]
+
+        return autograd.builtins.list(params)
+
+    @autograd.extend.primitive
+    def _execute(self, params, device):
+        params = autograd.builtins.tuple(params)
+
+        res = self.execute_device(params, device=device)
+
+        if res.dtype == np.dtype("object"):
+            return np.hstack(res)
+
+        requires_grad = False
+
+        if self.trainable_params:
+            requires_grad = True
+
+        return np.array(res, requires_grad=requires_grad)
+
+    @staticmethod
+    def vjp(ans, self, params, device):  # pylint: disable=unused-argument
+        """Returns the vector-Jacobian product operator for the quantum tape.
+
+        Takes the same arguments as :meth:`~.execute`, plus `ans`.

```

Comment:
Better to put in a full list of args

Reply:
lol I was thinking the same. I copied this originally from https://pennylane.readthedocs.io/en/stable/_modules/pennylane/interfaces/autograd.html#to_autograd

Reply:
Who wrote that? :)

Reply:
Turns out it was me 😆 . Fixed.